### ISSUE TITLE
Future Work

### ISSUE THREAD

**Issue by tiffanyfong** [tiffanyfong/PoiNotation#1]
**Language related**
- [ ] Prevent repeated properties (medium)
- [ ] Add `petals: Int` to syntax/IR (medium)
  - Note: Ensure that `petals` and `rotations` don't conflict. Petals have different meanings depending on `handleSpin` (particularly `inspin` and `antispin`)
- [ ] Implement variables in syntax (difficult)
  - Users can give names to poi moves for reuse: 
```
ccwExtension = {extended: true, armSpin: ccw, handleSpin: ccw, rotations: 1}
ccwExtension2 = ccwExtension * 2
```
- [ ] Implement variants in syntax (difficult)
  - Properties are added onto a non-default move:
```
threePetalFlower = {extended: true, handleSpin: antispin, rotations: 2}
fourPetalFlower = threePetalFlower.variant(rotations: 3)
```
- [ ] Make choreography for two poi (difficult)
  - [ ] Add TwoPoiMoves to the IR for moves with two simultaneous poi (medium)
- [ ] Error-handling from IR to simulator: include variable names in error messages (easy, but only after variables are implemented)
- [ ] Add more properties for poi moves

**System-related**
- [ ] Add documentation (easy)
- [ ] Incorporate JavaScript for the IR -> Simulator aspect, so that users don't need to copy and paste output (difficult)
- [ ] Web app instead of command-line interface, or at least provide both (difficult)
